sys/linux/dev_dri.txt

```diff
@@ -551,11 +566,18 @@ drm_control {
 	irq	int32
 }
 
-drm_agp_buffer {
-	size	intptr
-	handle	drm_agp_handle[opt]
-	type	flags[drm_agp_mem_type, intptr]
-	physic	intptr
+drm_agp_buffer$DRM_IOCTL_AGP_ALLOC {
+	size	intptr	(in)
+	handle	drm_agp_handle	(out)
+	type	flags[drm_agp_mem_type, intptr]	(in)
+	physic	intptr	(out)
+}
+
+drm_agp_buffer$DRM_IOCTL_AGP_FREE {
+	size	const[0, intptr]
+	handle	drm_agp_handle	(in)
```

Comment:
Again, this whole struct is already `in`:
```
ioctl$DRM_IOCTL_AGP_FREE(fd fd_dri, cmd const[DRM_IOCTL_AGP_FREE], arg ptr[in, drm_agp_buffer$DRM_IOCTL_AGP_FREE])
```

Reply:
As above, this change is now taken back. ([here](DRM_IOCTL_AGP_FREE))